pyccel/codegen/printing/ccode.py

```diff
@@ -1024,7 +1024,7 @@ def function_signature(self, expr, print_arg_names = True):
         else:
             arg_code = ', '.join('{}'.format(self.function_signature(i.var, False))
                         if isinstance(i.var, FunctionAddress)
-                        else '{0}'.format(self.get_declare_type(i.var)) + (i.name if print_arg_names else '')
+                        else '{1}{0}'.format(self.get_declare_type(i.var), 'const ' if i.var.is_const else '') + (i.name if print_arg_names else '')
                         for i in args)
```

Comment:
I find this statement quite hard to read. It is really big! Can we break it up into more readable pieces?
E.g:
```
dec_types = [.. for i in args]
arg_code_list = [d + i.name for d,i in zip(dec_types, args)]
arg_code = ', '.join(..)
```

Reply:
makes more sense, I ll make the changes necessary and push it again